pipelined/bevy_ui2/src/ui_node.rs

```diff
@@ -0,0 +1,288 @@
+    FlexStart,
+    FlexEnd,
+    Center,
+    Stretch,
+    SpaceBetween,
+    SpaceAround,
+}
+
+impl Default for AlignContent {
+    fn default() -> AlignContent {
+        AlignContent::Stretch
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, Debug, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum Direction {
+    Inherit,
+    LeftToRight,
+    RightToLeft,
+}
+
+impl Default for Direction {
+    fn default() -> Direction {
+        Direction::Inherit
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, Debug, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum Display {
+    Flex,
+    None,
+}
+
+impl Default for Display {
+    fn default() -> Display {
+        Display::Flex
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, Debug, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum FlexDirection {
+    Row,
+    Column,
+    RowReverse,
+    ColumnReverse,
+}
+
+impl Default for FlexDirection {
+    fn default() -> FlexDirection {
+        FlexDirection::Row
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, Debug, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum JustifyContent {
+    FlexStart,
+    FlexEnd,
+    Center,
+    SpaceBetween,
+    SpaceAround,
+    SpaceEvenly,
+}
+
+impl Default for JustifyContent {
+    fn default() -> JustifyContent {
+        JustifyContent::FlexStart
+    }
+}
+
+// TODO: add support for overflow settings
+// #[derive(Copy, Clone, PartialEq, Debug)]
+// pub enum Overflow {
+//     Visible,
+//     Hidden,
+//     Scroll,
+// }
+
+// impl Default for Overflow {
+//     fn default() -> Overflow {
+//         Overflow::Visible
+//     }
+// }
+
+#[derive(Copy, Clone, PartialEq, Debug, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum PositionType {
+    Relative,
+    Absolute,
+}
+
+impl Default for PositionType {
+    fn default() -> PositionType {
+        PositionType::Relative
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, Debug, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum FlexWrap {
+    NoWrap,
+    Wrap,
+    WrapReverse,
+}
+
+impl Default for FlexWrap {
+    fn default() -> FlexWrap {
+        FlexWrap::NoWrap
+    }
+}
+
+#[derive(Component, Default, Copy, Clone, Debug, Reflect)]
+#[reflect(Component)]
+pub struct CalculatedSize {
+    pub size: Size,
+}
+
+#[derive(Component, Default, Copy, Clone, Debug, Reflect)]
+#[reflect(Component)]
+pub struct UiColor(pub Color);
+
+impl From<Color> for UiColor {
+    fn from(color: Color) -> Self {
+        Self(color)
+    }
+}
+
+// NOTE: Option<Handle<Image>> is reflected as a value, so this derive(Reflect) requires Handle<T> to derive Serialize and Deserialize
+#[derive(Component, Default, Clone, Debug, Reflect)]
+#[reflect(Component)]
+pub struct UiImage(pub Option<Handle<Image>>);
```

Comment:
I'd like to suggest we use just an `Handle<Image>` without an option, and use a 1x1 image as the default for the handle. Alternatively, we could make the handle a weak handle that points to nothing.

Reply:
A default handle for images was just added in #3270, I'll convert this PR to use it too.

Reply:
Yeah I like this

Reply:
Done in 67ae82e